diff --git a/definitions.h b/definitions.h
index a380a3a..f4463bf 100644
--- a/definitions.h
+++ b/definitions.h
@@ -1,4 +1,4 @@
-#define WORKSPACES      10
+#define WORKSPACES      5
 #define BUTTONMASK      XCB_EVENT_MASK_BUTTON_PRESS|XCB_EVENT_MASK_BUTTON_RELEASE
 #define NET_WM_FIXED    0xffffffff  // Value in WM hint which means this window is fixed on all workspaces.
 #define TWOBWM_NOWS     0xfffffffe  // This means we didn't get any window hint at all.
